packages/design-picker/src/components/style-variation-badges/utils.ts

```diff
@@ -0,0 +1,68 @@
+import type {
+	StyleVariation,
+	StyleVariationSettingsColorPalette,
+	StyleVariationStylesColor,
+	StyleVariationPreview,
+	StyleVariationPreviewColorPalette,
+} from '../../types';
+
+const COLOR_PALETTE_SUPPORTS = [ 'background', 'foreground', 'primary', 'secondary', 'tertiary' ];
+
+const GLOBAL_STYLES_SUPPORTS: {
+	style: keyof StyleVariationStylesColor;
+	infix: 'color';
+	mapTo: keyof StyleVariationPreviewColorPalette;
+}[] = [
+	{ style: 'background', infix: 'color', mapTo: 'background' },
+	{ style: 'text', infix: 'color', mapTo: 'foreground' },
+];
+
+export function getPreviewStylesFromVariation(
+	variation: StyleVariation,
+	coreColors?: StyleVariationPreviewColorPalette
+): StyleVariationPreview {
+	const globalColors = {
+		...getValueFromVariationStylesColor( variation ),
+		...coreColors,
+	};
+
+	let color: StyleVariationPreviewColorPalette = {};
+	for ( const key of COLOR_PALETTE_SUPPORTS ) {
+		const value = getValueFromVariationSettingColorPalette( variation, key );
+		color = { ...color, ...( value && { [ key ]: value } ) };
+	}
+
+	// Check for global styles, they take precedence.
+	if ( globalColors ) {
```

Comment:
This will be always true, as an empty object (`{}`) is truthy. I think we can just remove this `if` condition altogether.

Reply:
Okay now this is not true anymore :)

Reply:
Way ahead of me 🙂